torch/distributed/algorithms/ddp_comm_hooks/__init__.py

```diff
@@ -38,8 +39,23 @@ def register_ddp_comm_hook(
         to the DDP model. User can specify the type of hook as an enum
         ``DDPCommHookType`` type using ``comm_hook_type`` input. State input will
         be passed to the model.
+        Uses Python comm hook implementations.
 
         Example::
             >>> register_ddp_comm_hook(DDPCommHookType.FP16_COMPRESS, model, state)
     """
     comm_hook_type.value(model=model, state=state)
+
+
+def register_builtin_ddp_comm_hook(comm_hook_type: dist.BuiltinCommHookType, model: DistributedDataParallel):
+    """
+        Registers the hooks of ``torch/csrc/distributed/c10d/default_comm_hooks.h``
+        to the DDP model. User can specify the type of hook as an enum
+        ``dist.BuiltinCommHookType`` type using ``comm_hook_type`` input.
+        Uses C++ comm hook implementations.
+
+        Example::
+            >>> register_builtin_ddp_comm_hook(dist.BuiltinCommHookType.FP16_COMPRESS, model)
+    """
+
+    model.reducer._register_builtin_comm_hook(comm_hook_type)
```

Comment:
Why do we have this additional API? Isn't the API in DDP sufficient?

Reply:
You are totally right! It was created when I was thinking of passing the hook type as a pyobject earlier to be called from Python API. Forgot to delete it later. Good catch!